tests/test_dia_calculation_plugins.py

```diff
@@ -190,33 +294,42 @@ def testCalculate(self):
         """Test mean value calculation.
         """
         n_sources = 10
-        diaObject = dict()
-        diaSources = pd.DataFrame(data={"psFlux": np.linspace(-1, 1, n_sources),
-                                        "psFluxErr": np.ones(n_sources)})
+        objId = 0
+
+        # Test expected mean.
+        diaObjects = pd.DataFrame({"diaObjectId": [objId]})
+        diaSources = pd.DataFrame(
+            data={"diaObjectId": n_sources * [objId],
+                  "filterName": n_sources * ["u"],
+                  "diaSourceId": np.arange(n_sources, dtype=int),
+                  "psFlux": np.linspace(-1, 1, n_sources),
+                  "psFluxErr": np.ones(n_sources)})
 
         plug = WeightedMeanDiaPsFlux(WeightedMeanDiaPsFluxConfig(),
                                      "ap_meanFlux",
                                      None)
-        plug.calculate(diaObject, diaSources, diaSources, "u")
-
-        self.assertAlmostEqual(diaObject["uPSFluxMean"], 0.0)
-        self.assertAlmostEqual(diaObject["uPSFluxMeanErr"], np.sqrt(1 / n_sources))
-        self.assertEqual(diaObject["uPSFluxNdata"], n_sources)
-
-        diaObject = dict()
-        plug.calculate(diaObject, [], [], "g")
+        run_multi_plugin(diaObjects, diaSources, "u", plug)
 
-        self.assertTrue(np.isnan(diaObject["gPSFluxMean"]))
-        self.assertTrue(np.isnan(diaObject["gPSFluxMeanErr"]))
-        self.assertEqual(diaObject["gPSFluxNdata"], 0)
+        self.assertAlmostEqual(diaObjects.loc[objId, "uPSFluxMean"], 0.0)
+        self.assertAlmostEqual(diaObjects.loc[objId, "uPSFluxMeanErr"],
+                               np.sqrt(1 / n_sources))
+        self.assertEqual(diaObjects.loc[objId, "uPSFluxNdata"], n_sources)
 
-        diaObject = dict()
-        diaSources.loc[4, "psFlux"] = np.nan
-        plug.calculate(diaObject, diaSources, diaSources, "r")
+        # Test expected mean with a nan value.
+        diaObjects = pd.DataFrame({"diaObjectId": [objId]})
+        fluxes = np.linspace(-1, 1, n_sources)
+        fluxes[4] = np.nan
+        diaSources = pd.DataFrame(
+            data={"diaObjectId": n_sources * [objId],
+                  "filterName": n_sources * ["r"],
+                  "diaSourceId": np.arange(n_sources, dtype=int),
+                  "psFlux": fluxes,
+                  "psFluxErr": np.ones(n_sources)})
+        run_multi_plugin(diaObjects, diaSources, "r", plug)
 
```

Comment:
test a non-zero value

Reply:
Realized there was a bug in the mean calculation where totWeight would be incorrect if a flux was nan but not the err. Added a test for both weighted means.